core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -345,19 +354,27 @@ public static String deletePropertiesStatement(Set<String> properties) {
 
   static boolean namespaceExists(
       String catalogName, JdbcClientPool connections, Namespace namespace) {
+
+    String namespaceEquals = JdbcUtil.namespaceToString(namespace);
+    // when namespace has sub-namespace then additionally checking it with LIKE statement.
+    // catalog.db can exists as: catalog.db.ns1 or catalog.db.ns1.ns2
+    String namespaceStartsWith =
+        namespaceEquals.replace("\\", "\\\\").replace("_", "\\_").replace("%", "\\%") + ".%";
```

Comment:
Since this converts `\` into an escaped character, I think we also need test cases for namespaces with `\`.

Reply:
added one test